app/controllers/WhatDoYouWantToDoController.scala

```diff
@@ -120,12 +118,18 @@ class WhatDoYouWantToDoController @Inject()(
 
   private def allowWhatDoYouWantToDo(implicit request: Request[AnyContent], user: AuthedUser): Future[Result] = {
     val nino = user.nino
-    for {
-      hasTaxCodeChanged <- taxCodeChangeService.hasTaxCodeChanged(nino)
-      showJrsTile       <- jrsService.checkIfJrsClaimsDataExist(nino)
-      model             <- whatToDoView(nino, hasTaxCodeChanged, showJrsTile)
-      _                 <- auditNumberOfTaxCodesReturned(nino, showJrsTile)
-    } yield Ok(whatDoYouWantToDoTileView(WhatDoYouWantToDoForm.createForm, model, applicationConfig))
+
+    taxCodeChangeService.hasTaxCodeChanged(nino) flatMap {
+      case Right(taxCodeChanged) =>
+        for {
+          showJrsTile <- jrsService.checkIfJrsClaimsDataExist(nino)
+          model       <- whatToDoView(nino, taxCodeChanged, showJrsTile)
+          _           <- auditNumberOfTaxCodesReturned(nino, showJrsTile)
+        } yield Ok(whatDoYouWantToDoTileView(WhatDoYouWantToDoForm.createForm, model, applicationConfig))
```

Comment:
can we do a match on the yield so we can cover all possible error cases?

Reply:
Not sure I get that but can discuss this later on the call @brendanstraughan